src/contour/TerminalWindow.cpp

```diff
@@ -67,6 +70,26 @@ namespace contour
 
 using actions::Action;
 
+static auto loadSessionFile(FileSystem::path sessionFilePath)
```

Comment:
I think it would make sense to even create a tiny struct here insteaad of returning a 3-tuple with each element type being the same. That could be easily mixed up accidentally. What about:

```cpp
struct SessionState
{
    std::string configPath;
    std::string profileName;
    std::string gridBuffer;
};
```